### PR TITLE
Database tests

### PR DESCRIPTION
**Pull request by KiviluomaNeera**
Test classes for diagnosisRepository and DataHandler.

DiagnosisRepositoryTest tests that the data structure stays correct after its fetched from database and that there won't be any null values which would suggest that there is problems with fetching the data.

DataHandlerTest mainly tests that the link between repositories and dataHandler works and exceptions work as wanted.

### CODE REVIEW

File: backend/antibiootit/src/test/java/fi/tuni/koodimankelit/antibiootit/database/DiagnosisRepositoryTest.java [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+        assertEquals(1, antibiotics.size());
+        Antibiotic antibiotic = antibiotics.get(0);
+        assertEquals("", antibiotic.getAntibiotic());
+        assertEquals(0, antibiotic.getDays());
+        assertEquals(0, antibiotic.getStrength().size());
+        assertEquals(0, antibiotic.getDoseMultipliers().size());
+    }
+
+    @Test
+    public void testFindDiagnosisByIdHasAntibiotics() {
```
Comment (pampula) [inline backend/antibiootit/src/test/java/fi/tuni/koodimankelit/antibiootit/database/DiagnosisRepositoryTest.java:67]: Could this be divided into smaller tests?

File: backend/antibiootit/src/test/java/fi/tuni/koodimankelit/antibiootit/database/DiagnosisRepositoryTest.java [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+        testMixture(mixture);
+        testTablet(tablet);
+
+    }
+
+    @Test
+    public void testGetAllDiagnosisInfos() {
+        List<DiagnosisInfo> result = diagnosisRepository.getAllDiagnosisInfos();
+
+        // Test that there is right amount of diagnosis and that the order is correct
```
Comment (pampula) [inline backend/antibiootit/src/test/java/fi/tuni/koodimankelit/antibiootit/database/DiagnosisRepositoryTest.java:175]: What if the contents of the database gets changed/updated?

**Review by pampula — CHANGES_REQUESTED** [review]
Seems okay, but could a mocking be used in checking if the repo returns all diagnosis infos? Now is the database's contents are updated, the test is no more valid.

**Review by juhor00 — CHANGES_REQUESTED** [review]
Eveliina is kind of correct. In DataHandlerTest, you should mock the repository service. You are just  testing the handler implementation there. Eveliina commented that under the repository tests, which should not be mocked.
The repository tests should not use the production data, though. ChatGPT suggested setting up in-memory MongoDB and populating test data for the tests. This way, the tests are not tied to production database (which could change or connection fail etc.). Here's an example implementation for Repository tests:
image

In summary, the DataHandler tests should mock the repository, and the repository should use in-memory database instance instead of the production one deployed in hosting service. This is at least the case for unit tests, later when we test the integrations and system altogether, it could be another way.
